modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -2507,6 +2521,79 @@ public void testGetObjectEntriesWithAggregationFacets() throws Exception {
 		Assert.assertEquals(facetValue.getNumberOfOccurrences(), (Integer)2);
 	}
 
+	@Test
+	public void testGetObjectEntryDocument() throws Exception {
+		_objectEntryLocalService.addObjectEntry(
+			adminUser.getUserId(), 0,
+			_objectDefinition1.getObjectDefinitionId(),
+			HashMapBuilder.<String, Serializable>put(
+				"externalReferenceCode", "newExternalReferenceCode"
+			).put(
+				"textObjectFieldName", StringUtil.randomId()
+			).put(
+				"textObjectFieldNameExtension", StringUtil.randomId()
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		_objectEntryLocalService.addObjectEntry(
+			adminUser.getUserId(), 0,
+			_objectDefinition2.getObjectDefinitionId(),
+			HashMapBuilder.<String, Serializable>put(
+				_objectRelationshipERCObjectFieldName,
+				"newExternalReferenceCode"
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		SearchRequestBuilder searchRequestBuilder =
+			_searchRequestBuilderFactory.builder();
+
+		searchRequestBuilder.entryClassNames(
+			_objectDefinition1.getClassName()
+		).emptySearchEnabled(
+			true
+		).withSearchContext(
+			searchContext -> {
+				searchContext.setAttribute(
+					Field.STATUS, WorkflowConstants.STATUS_ANY);
+				searchContext.setAttribute(
+					"objectDefinitionId",
+					_objectDefinition1.getObjectDefinitionId());
+				searchContext.setCompanyId(_objectDefinition1.getCompanyId());
+
+				searchContext.setGroupIds(new long[] {0});
+			}
+		);
+
+		SearchResponse searchResponse = _searcher.search(
+			searchRequestBuilder.build());
+
+		SearchHits searchHits = searchResponse.getSearchHits();
+
+		SearchHit searchHit = searchHits.getSearchHits(
+		).get(
+			0
+		);
+
+		Document document = searchHit.getDocument();
+
+		JSONObject indexedValuesJSONObject = JSONFactoryUtil.createJSONObject(
+			"{" + document.getString("objectEntryContent") + "}");
+
+		for (ObjectField objectField :
+				objectFieldLocalService.getObjectFields(
+					_objectDefinition1.getObjectDefinitionId(), false)) {
+
+			if (objectField.isIndexed()) {
+				Assert.assertTrue(
+					indexedValuesJSONObject.has(objectField.getName()));
+			}
+			else {
+				Assert.assertFalse(
+					indexedValuesJSONObject.has(objectField.getName()));
+			}
+		}
+	}
```

Comment:
Hey @p-albuquerque could you please check if it can be simplified to this:

```java
	@Test
	public void testGetObjectEntryDocument() throws Exception {
		_objectEntryLocalService.addObjectEntry(
			adminUser.getUserId(), 0,
			_objectDefinition1.getObjectDefinitionId(),
			HashMapBuilder.<String, Serializable>put(
				"textObjectFieldName", StringUtil.randomId()
			).build(),
			ServiceContextTestUtil.getServiceContext());

		SearchResponse searchResponse = _searcher.search(
			_searchRequestBuilderFactory.builder(
			).entryClassNames(
				_objectDefinition1.getClassName()
			).emptySearchEnabled(
				true
			).withSearchContext(
				searchContext -> {
					searchContext.setAttribute(
						"objectDefinitionId",
						_objectDefinition1.getObjectDefinitionId());
					searchContext.setAttribute(
						Field.STATUS, WorkflowConstants.STATUS_ANY);
					searchContext.setCompanyId(
						_objectDefinition1.getCompanyId());
					searchContext.setGroupIds(new long[] {0});
				}
			).build()
		);

		SearchHits searchHits = searchResponse.getSearchHits();

		List<SearchHit> searchHitList = searchHits.getSearchHits();

		SearchHit searchHit = searchHitList.get(0);

		Document document = searchHit.getDocument();

		JSONObject objectEntryContentJSONObject =
			JSONFactoryUtil.createJSONObject(
				"{" + document.getString("objectEntryContent") + "}");

		for (ObjectField objectField :
				objectFieldLocalService.getObjectFields(
					_objectDefinition1.getObjectDefinitionId(), false)) {

			Assert.assertEquals(
				objectField.isIndexed(),
				objectEntryContentJSONObject.has(objectField.getName()));
		}
	}
```

Reply:
Wow, I like this way of the assert!
Thanks

But I can't do that:
```java
		_objectEntryLocalService.addObjectEntry(
			adminUser.getUserId(), 0,
			_objectDefinition1.getObjectDefinitionId(),
			HashMapBuilder.<String, Serializable>put(
				"textObjectFieldName", StringUtil.randomId()
			).build(),
			ServiceContextTestUtil.getServiceContext());
```

Because, even if I set "testObjectFieldName" and "extensionObjectFieldName" indexed, 
I need to fill some value on this fields, otherwise the objectEntryContent cames without them.